src/trait_/common_ord.rs

```diff
@@ -49,6 +59,13 @@ pub fn build_ord_signature(item: &Item, trait_: &DeriveTrait, body: &TokenStream
 				}
 			};
 
+			let incomparable = build_incomparable_pattern(variants).into_iter();
+			let incomparable = quote! {
+				#(if ::core::matches!(self, #incomparable) || ::core::matches!(__other, #incomparable) {
+					return ::core::option::Option::None
+				})*
+			};
```

Comment:
I just used an if statement with two `matches!` do you think there is a better implementation for the default case with multiple variants?

Reply:
Can't think of doing anything better then `matches!(...) || matches!(..)`, but instead of making multiple if statements you could just put them all in one, you just have to add a check beforehand to make sure there are any incomparables at all.

Reply:
oh that should generate a single one (still need to add a test) `incomparable` is an `Option::into_iter`

Reply:
Oh! Right! Perfect then.